services/common/src/main/java/org/finra/gatekeeper/common/authfilter/parser/SSOParser.java

```diff
@@ -18,25 +18,45 @@
 package org.finra.gatekeeper.common.authfilter.parser;
 
 import javax.servlet.http.HttpServletRequest;
+import java.util.Collections;
+import java.util.HashSet;
 import java.util.Optional;
+import java.util.Set;
 
 public class SSOParser implements UserParser {
 
     private final String userIdHeader;
+    private final String groupsHeader;
+    private final String emailHeader;
+    private final String userNameHeader;
```

Comment:
Why not just define a different SSOParser for the header method that will look for the extra headers?